### ISSUE TITLE
Can't do a fresh pip install from source distribution: pybind11 issues.

### ISSUE THREAD

**Issue by JustinTArthur** [inducer/pyopencl#278]
On macOS, 10.12.6 w/ a fresh Python 3.7 virtualenv with pip 19.0.3, `pip install pyopencl` must download the 2018.2.3 source package, given that none of the 2018.2.4 manylinux wheels match.'

The setup script ultimately fails due to pybind11 not being found. It looks like it's being included via `setup_requires` which uses easy_install if I recall. If I `easy_install pybind11` beforehand, `check_pybind11()` passes, but the missing header error still happens in install.  Only if I `pip install pybind11` beforehand does the install succeed.
```
$ pip install pyopencl
Looking in indexes: https://pypi.org/simple
Collecting pyopencl
  Using cached https://files.pythonhosted.org/packages/f8/2a/953d7373893f0e2e4f48db996712a5613dccb80bc9339d51d8437c6e21c4/pyopencl-2018.2.3.tar.gz
Collecting numpy (from pyopencl)
  Downloading https://files.pythonhosted.org/packages/a6/6f/cb20ccd8f0f8581e0e090775c0e3c3e335b037818416e6fa945d924397d2/numpy-1.16.2-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (13.9MB)
    100% |████████████████████████████████| 13.9MB 1.9MB/s 
Collecting pytools>=2017.6 (from pyopencl)
  Downloading https://files.pythonhosted.org/packages/ac/a3/f54f7190315ad41b7334d8733350e7fcefded8f25e0b45e2329b80279921/pytools-2019.1.tar.gz (57kB)
    100% |████████████████████████████████| 61kB 9.0MB/s 
Collecting decorator>=3.2.0 (from pyopencl)
  Downloading https://files.pythonhosted.org/packages/f1/cd/7c8240007e9716b14679bc217a1baefa4432aa30394f7e2ec40a52b1a708/decorator-4.3.2-py2.py3-none-any.whl
Collecting appdirs>=1.4.0 (from pyopencl)
  Downloading https://files.pythonhosted.org/packages/56/eb/810e700ed1349edde4cbdc1b2a21e28cdf115f9faf263f6bbf8447c1abf3/appdirs-1.4.3-py2.py3-none-any.whl
Collecting six>=1.9.0 (from pyopencl)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Building wheels for collected packages: pyopencl, pytools
  Building wheel for pyopencl (setup.py) ... error
  Complete output from command /Users/jarthur/.virtualenvs/ocl_venv/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/private/var/folders/04/5htmmmqn4px5y6wn40x8wp780000gn/T/pip-install-hs9jrdop/pyopencl/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /private/var/folders/04/5htmmmqn4px5y6wn40x8wp780000gn/T/pip-wheel-qvqwlarf --python-tag cp37:
  ---------------------------------------------------------------------------
  Pybind11 is not installed.
  ---------------------------------------------------------------------------
  Very likely, the build process after this message will fail.
```

**Comment by JustinTArthur** [comment]
Looks like it's a known issue, according to pybind/python_example#16.

**Comment by inducer** [comment]
Yup. I'm not sure I'm eager to put in engineering effort to work around what ultimately seems to be a pip deficiency.

**Comment by gmagno** [comment]
@JustinTArthur I am confident `pipenv` (as opposed to pip) can deal with that issue.

**Comment by isuruf** [comment]
It's not pip's fault, it's pybind11's. (numpy works fine in setup_requires)

**Comment by JustinTArthur** [comment]
Upstream issue pybind/pybind11#1067 was fixed with pybind/pybind11#1995 and is finally available in pybind11>=2.5.0 on PyPI. PyOpenCL should be pip-installable again.
